docs/proposed/recovery.rst

```diff
@@ -0,0 +1,97 @@
+.. -*- coding: utf-8 -*-
+
+.. _snapshots:
+
+Audience
+========
+
+This document is aimed at programmers working on integrating with magic-folder.
+
+
+Motivation
+==========
+
+It is useful to be able to back up enough information in order to recover a magic-folder on a new device.
+This document describes both the process of gathering the informatino to back up and using it to re-create a similar magic-folder.
+
+
+Overview
+========
+
+A magic-folder consists of a Collective DMD containing sub-directories pointing at the read-capabilities of one or more participant's Personal DMDs.
+If this magic-folder instance is the "admin" instance, it will contain a write-capability for the Collective.
+It will also contain a write-capablity corresponding to the Personal DMD.
+
+The rough process of "recovery" is:
+
+- create a new, fresh Magic Folder on the new instance
+- add the "recovery" instance as a participant in this new Magic Folder
+
+The new instance's local magic-path will fill up with all the files in the old instance as the downloader gets to them.
+In a usual case of "recovery" from e.g. a lost or broken device, this participant would not produce new Snapshots.
+However, if it does start producing new Snapshots the Magic Folder will continue to synchronize and detect conflicts as normal.
+
+**Note:** this document is only concerned with single-participant Magic Folders for now and does not cover recovery of other participants.
+
+
+Gathering the Information
+-------------------------
+
+In order to recover a Magic Folder on a new device you will need some information about it:
+
+- the Collective DMD (either write-capability or read-capability)
+- the Personal DMD (usually a write-capability but you could choose to only save the read-capability corresponding to it)
+
+This can be gathered for all folders at once with a ``magic-folder`` command::
+
+    $ magic-folder list --json --include-secret-information
+
+Running that command will produce a JSON ``dict`` containing one key for each Magic Folder.
+These will point to a ``dict`` containing all the information about this Magic Folder.
+The relevant keys here are:
+- ``"upload-dircap"``: our Personal DMD write-capability
+- ``"collective-dircap"``: a write-capability or read-capability to the Collective DMD (not used in this revision of this document but it is useful to save)
+
+You could choose to keep _all_ of the JSON from the ``magic-folder list`` incantation.
```

Comment:
Given my comment above about read vs write capabilities, I wonder if we should have a command and/or option to store all the relevant data with *just* the read-capabilities. And maybe also omitting the local file paths.

Will gridsync uses this, or does it do its own thing of storing that data in its own tahoe directory?

Reply:
GridSync stores these data itself, somewhere. I think from magic-folders perspective it makes the most sense to dump the information we have as straight as possible. This lets integrators like GridSync do what they deem appropriate.

It could make sense at some point to add an _option_ to spit out only read-caps .. or e.g. a utility command like `magic-folder-api to-read-cap` or something. For now, no action needed until we have more guidance from at least GridSync on what it wants to store..?

Reply:
Well, we should perhaps have a conversation about @crwood about how we want this implemented between magic-folder and gridsync, and reflect that here.

Reply:
Okay .. can we just file tickets against `magic-folder list` if what that outputs currently isn't sufficient for GridSync's needs?